homeassistant/components/media_player/mediaroom.py

```diff
@@ -81,12 +81,20 @@ def callback_notify(notify):
     if not config[CONF_OPTIMISTIC]:
         from pymediaroom import install_mediaroom_protocol
 
-        already_installed = hass.data.get(DISCOVERY_MEDIAROOM, False)
+        already_installed = hass.data.get(DISCOVERY_MEDIAROOM, None)
         if not already_installed:
-            await install_mediaroom_protocol(
+            hass.data[DISCOVERY_MEDIAROOM] = await install_mediaroom_protocol(
                 responses_callback=callback_notify)
+
+            async def stop_discovery(event):
```

Comment:
Should this not be just a `@callback` imported from homeassistant.core?

Reply:
didn't get that memo 😄 

Fixed!